app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/websocket/WebSocketServerHandler.kt

```diff
@@ -29,43 +31,49 @@ class WebSocketServerHandler(
 
         compositeDisposable.clear()
         server = CustomWebSocketServer(SERVER_PORT,
-                onMessageReceived = { webSocket, message -> handleMessage(webSocket, message) }
+            onMessageReceived = { webSocket, message -> handleMessage(webSocket, message) }
         ).apply {
             startServer().subscribeOn(Schedulers.io())
-                    .subscribeBy(
-                            onComplete = {
-                                Timber.i("CustomWebSocketServer started")
-                            },
-                            onError = { e ->
-                                restartServer()
-                                Timber.e("launch failed $e")
-                            }
-                    ).addTo(compositeDisposable)
+                .subscribeBy(
+                    onComplete = {
+                        Timber.i("CustomWebSocketServer started")
+                    },
+                    onError = { e ->
+                        restartServer()
+                        Timber.e("launch failed $e")
+                    }
+                ).addTo(compositeDisposable)
             connectedClients()
-                    .subscribeOn(Schedulers.io())
-                    .observeOn(AndroidSchedulers.mainThread())
-                    .subscribeBy(onNext = { connections ->
-                        clientConnectionStatus.postValue(
-                                if (connections > 0) ClientConnectionStatus.CLIENT_CONNECTED
-                                else ClientConnectionStatus.EMPTY)
-                    })
-                    .addTo(compositeDisposable)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .doOnNext { connections ->
+                    val status = if (connections > 0) {
+                        ClientConnectionStatus.CLIENT_CONNECTED
+                    } else {
+                        ClientConnectionStatus.EMPTY
+                    }
+                    publishSubjectClientConnectionStatus.onNext(status)
+                }
+                .subscribeBy(onError = {
+                    Timber.e(it)
+                })
+                .addTo(compositeDisposable)
         }
     }
 
-    fun restartServer() {
+    private fun restartServer() {
         stopServer()
         Completable.timer(5, TimeUnit.SECONDS)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/f1e62ee67a21a21873bbc26c24839e1cf2930e4c/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/websocket/WebSocketServerHandler.kt#L66"></span>This expression contains a magic number. Consider defining it to a well named constant.</td>
    </tr>
  </tbody>
</table>